src/scenarios/shared/runner.py

```diff
@@ -327,7 +329,7 @@ def run(self):
                     getLogger().info("Removed: " + os.path.join(const.TRACEDIR, file))
                     os.remove(file)
 
-            cmdline = xharnesscommand() + [self.devicetype, 'state', '--adb']
+            cmdline = xharnesscommand() + ['android', 'state', '--adb']
```

Comment:
Is there a reason that we are moving to the hardcoded value here, when we know that the value will be android because of the elif block?

Reply:
basically because of that, it is always `android` in that block so using a variable is just slightly harder to read IMO.

Reply:
Also `apple/wasm state --adb` is not a valid command.

That being said, you could now utilize the `android adb` command directly and this call and some other below can be removed. You don't need to now get ADB's location from `adb.stdout.strip()` and call
```
xharnesscommand() + ['android', 'adb', '--', 'shell', 'vm', 'size']
```

I'd even maybe add `adbcommand()` that would do `xharnesscommand() + ['android', 'adb', '--']`